aya/src/obj/mod.rs

```diff
@@ -510,13 +511,27 @@ fn parse_map(section: &Section, name: &str) -> Result<Map, ParseError> {
 }
 
 fn parse_map_def(name: &str, data: &[u8]) -> Result<bpf_map_def, ParseError> {
-    if mem::size_of::<bpf_map_def>() > data.len() {
+    if data.len() > mem::size_of::<bpf_map_def>() || data.len() < MINIMUM_MAP_SIZE {
         return Err(ParseError::InvalidMapDefinition {
             name: name.to_owned(),
         });
     }
 
-    Ok(unsafe { ptr::read_unaligned(data.as_ptr() as *const bpf_map_def) })
+    if data.len() < mem::size_of::<bpf_map_def>() {
+        let mut map_def = bpf_map_def::default();
+        unsafe {
+            let mut data_ptr = data.as_ptr();
+            let mut map_def_ptr = &mut map_def as *mut bpf_map_def as *mut u8;
+            for _ in 0..=data.len() {
```

Comment:
Instead of looping, please convert to byte slices and use `copy_from_slice`

Reply:
Fixed in 86b639b